rs/boundary_node/rate_limit_canister/interface.did

```diff
@@ -0,0 +1,89 @@
+type Version = nat64;   // Represents the config version number
+type Timestamp = nat64; // Represents timestamp in nanoseconds since the epoch (1970-01-01)
+type RuleId = text;     // Unique identifier for each rule
+
+
+// Input structure for defining a rule with mandatory fields within a config
+type InputRule = record {
+  id: RuleId;        // Unique identifier for the rule
+  rule_raw: blob;    // Raw rule data (in binary format), expected to be a valid json object
+  description: text; // Textual description of the rule
+};
+
+// Output structure for rules
+// Optional fields rule_raw and description may remain hidden while the rule is under confidentiality restrictions
+type OutputRule = record {
+  id: RuleId;                        // Unique identifier for the rule
+  rule_raw: opt blob;                // Raw rule data (in binary format), expected to be a valid json object, none if the rule is currently confidential
+  description: opt text;             // Textual description of the rule, none if the rule is currently confidential
+  disclosed_at: opt Timestamp;       // Timestamp when the rule was disclosed, none if the rule is still confidential
+  disclosed_in_version: opt Version; // Version in which the rule was disclosed, none if the rule is still confidential
+};
+
+type OutputConfig = record {
+  rules: vec OutputRule;
+};
+
+// Response structure for returning the requested configuration and associated metadata
+type OutputConfigResponse = record {
+    version: Version;         // Version of the configuration
+    active_since: Timestamp;  // Time when this configuration became added (became active)
+    config: OutputConfig;     // Contains the list of rules
+};
+
+// Verbose details of an individual rule
+// Optional rule_raw and description fields are for restricted publicly viewing access
+type OutputRuleMetadata = record {
+  id: RuleId;                      // Unique identifier for the rule
+  rule_raw: opt blob;              // Raw rule data (binary format), expected to be a valid json object, none if the rule is currently confidential
+  description: opt text;           // Textual description of the rule, none if the rule is currently confidential
+  disclosed_at: opt Timestamp;     // Timestamp when the rule was disclosed, none if the rule is still confidential
+  disclosed_in_version: Version;   // Version when the rule was disclosed, none if the rule is still confidential
```

Comment:
Only one of the two is needed. Drop the timestamp.

Reply:
this adds additional transparency for users; a version can last weeks/months, giving the exact time when the rule is exposed is a really transparent approach IMO

Reply:
I disagree: showing too much information (especially redundant) can hinder transparency as it distracts from the important information.

It seems for you the version is the important information: You only provide a timestamp for the `disclosed` information, but not for when the rule was added/removed. Is the time at which the rule was disclosed really that important? I would argue that it is much more important when the rule was actually active.

Reply:
> You only provide a timestamp for the disclosed information, but not for when the rule was added/removed.

This is also important and i do provide it [here](https://github.com/dfinity/ic/pull/1865/files#diff-c57930d3199313c72d0a8d5814baa996c9c4db7e30e4f23ed805a15515d04fc8R30).

>  Is the time at which the rule was disclosed really that important?
IMO it is important, if i was a user i'd like to see that yes.

Reply:
i think call to [get_config()](https://github.com/dfinity/ic/pull/1865/files#diff-c57930d3199313c72d0a8d5814baa996c9c4db7e30e4f23ed805a15515d04fc8R85) should aim at giving more important info, like rule and description, but call to `get_rule_by_id()` should contain a rather detailed info.

Reply:
Change my mind after our discussion: I would remove `disclosed_in_version`. Disclosing is not bound to a specific version, but just happens at some time. For the rule the "time" it was active matters.